src/data/resolvers/coinMachine.ts

```diff
@@ -211,5 +212,136 @@ export const coinMachineResolvers = ({
         return null;
       }
     },
+    async coinMachinePreviousSales(_, { colonyAddress }) {
```

Comment:
So while I'm generally ok with having a local resolver for stuff like this, this a **big one**.

There's too much going on, too many calls to the network being made, as well as too many events being fetched and parse.

Keeping in mind that sales are basically unlimited this has the potential of breaking the app _(by failing to load, due to the vast number crunching this resolver has to)_

So, as I said initially when we started Coin Machine, this needs to be put on the subgraph, and just fetch the sales data from there _(previous sales, current sale can still come from events)_

Talk to @area about ways to achieve this, otherwise this won't work in production.